kuryr_kubernetes/controller/handlers/lbaas.py

```diff
@@ -185,42 +185,49 @@ def on_present(self, endpoints):
-            except k_exc.K8sResourceNotFound:
-                # Note(yboaron) It's impossible to store neutron resources
-                # in K8S object since object was deleted. In that case
-                # we should rollback all neutron resources.
-                LOG.debug("LoadBalancerHandler failed to store Openstack "
-                          "resources in K8S object (not found)")
-                self.on_deleted(endpoints, lbaas_state)
+        try:
+            if self._sync_lbaas_members(endpoints, lbaas_state, lbaas_spec):
+                # Note(yboaron) For LoadBalancer services, we should allocate
+                # FIP, associate it to LB VIP and update K8S service status
+                if lbaas_state.service_pub_ip_info is None:
+                    service_pub_ip_info = (
+                        self._drv_service_pub_ip.acquire_service_pub_ip_info(
+                            lbaas_spec.type,
+                            lbaas_spec.lb_ip,
+                            lbaas_spec.project_id,
+                            lbaas_state.loadbalancer.port_id))
+                    if service_pub_ip_info:
+                        self._drv_service_pub_ip.associate_pub_ip(
+                            service_pub_ip_info,
+                            lbaas_state.loadbalancer.port_id)
+                        lbaas_state.service_pub_ip_info = service_pub_ip_info
+                        self._update_lb_status(
+                            endpoints,
+                            lbaas_state.service_pub_ip_info.ip_addr)
+                # REVISIT(ivc): since _sync_lbaas_members is responsible for
+                # creating all lbaas components (i.e. load balancer, listeners,
+                # pools, members), it is currently possible for it to fail (due
+                # to invalid Kuryr/K8s/Neutron configuration, e.g. Members' IPs
+                # not belonging to configured Neutron subnet or Service IP
+                # being in use by gateway or VMs) leaving some Neutron entities
+                # without properly updating annotation. Some sort of failsafe
+                # mechanism is required to deal with such situations (e.g.
+                # cleanup, or skip failing items, or validate configuration) to
+                # prevent annotation being out of sync with the actual Neutron
+                # state.
+                try:
+                    utils.set_lbaas_state(endpoints, lbaas_state)
+                except k_exc.K8sResourceNotFound:
+                    # Note(yboaron) It's impossible to store neutron resources
+                    # in K8S object since object was deleted. In that case
+                    # we should rollback all neutron resources.
+                    LOG.debug("LoadBalancerHandler failed to store Openstack "
+                              "resources in K8S object (not found)")
+                    self.on_deleted(endpoints, lbaas_state)
+        except k_exc.LoadBalancerRemoved as e:
```

Comment:
If this exception won't be re-raised is it worth to be handled as the overquota is handled?

https://github.com/openshift/kuryr-kubernetes/blob/release-3.11/kuryr_kubernetes/handlers/retry.py#L80-L83